### PR TITLE
S4 68 add business rules to api fixed

### PR DESCRIPTION
**Pull request by faggc**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/service/CompanyOfficerService.java [hunk 1]
```diff
@@ -27,11 +26,7 @@ public CompanyOfficerService(
         this.companyOfficerValidator = companyOfficerValidator;
     }
 
-    public boolean hasEnoughOfficersSelected(String companyNumber) {
-        final List<DissolutionDirector> selectedDirectors = repository.findByCompanyNumber(companyNumber)
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/service/CompanyOfficerService.java]: respository dependency can be removed

File: src/main/java/uk/gov/companieshouse/service/DissolutionValidator.java [hunk 2]
```diff
@@ -19,12 +20,18 @@ public DissolutionValidator(CompanyProfileService companyProfileService, Company
...
+    ) {
         if (!companyProfileService.isCompanyClosable(companyProfileApi)) {
             return Optional.of("Company must be of a closable type and have an active status");
         }
 
-        if (!companyOfficerService.hasEnoughOfficersSelected(companyProfileApi.getCompanyNumber())) {
+        if (!companyOfficerService.hasEnoughOfficersSelected(
+                companyProfileApi.getCompanyNumber(),
+                selectedDirectors
+        )) {
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/service/DissolutionValidator.java:34]: so many line breaks

File: src/main/java/uk/gov/companieshouse/service/DissolutionValidator.java [hunk 3]
```diff
@@ -19,12 +20,16 @@ public DissolutionValidator(CompanyProfileService companyProfileService, Company
...
+            CompanyProfileApi companyProfileApi,
+            List<DirectorRequest> selectedDirectors) {
         if (!companyProfileService.isCompanyClosable(companyProfileApi)) {
             return Optional.of("Company must be of a closable type and have an active status");
         }
 
-        if (!companyOfficerService.hasEnoughOfficersSelected(companyProfileApi.getCompanyNumber())) {
+        if (!companyOfficerService.hasEnoughOfficersSelected(
+                companyProfileApi.getCompanyNumber(),
+                selectedDirectors)) {
```
Comment (peadarkelly) [inline src/main/java/uk/gov/companieshouse/service/DissolutionValidator.java:32]: i don't know about you but i find this so hard to read compared to:
```
if (!companyOfficerService.hasEnoughOfficersSelected(companyProfileApi.getCompanyNumber(), selectedDirectors)) {
```

Comment (faggc) [reply src/main/java/uk/gov/companieshouse/service/DissolutionValidator.java:32]: that's fair,
I entered the line breaks for line length reasons>

Im not partial to them so ill get rid

**Pull request merged**